### PR TITLE
fix: Receipt requirement is counting old receipts

### LINKED ISSUE #299: bug (state): Receipt requirement is counting old receipts
# Current Behavior

After connecting Gmail and a retailer account, the receipt count requirement will be satisfied after the initial scans if there are at least 5 _past_ receipts discovered.

# Expected Behavior

The receipt requirement should only be satisfied after 5 _NEW_ receipts, not past receipts discovered during the initial scans.

### PR DESCRIPTION
**Pull request by MiroBenicio**
# Issues Addressed

Closes #299 

## Additional Information

I'm making this PR as a draft because Im not sure if this is the right way to solve the problem

so let me know what I can change or if its good

### CODE REVIEW

**Review by mike-audi — CHANGES_REQUESTED** [review]
We should not be adding another date into the mix. We should be using the start and end weeks that already exist. 

The first change that needs to happen is we need to set the correct receipt date on in the internal-handler code when we add the receipt to the store. 

The next challenge to solve is to determine the "start" week. This is where we really need **@timoguin to define how this is supposed to work.** Currently it uses a rolling filter by looking back at the previous 4 weeks. Once you meet all 4 criteria for 4 weeks straight, you get your reward.

What we need to know is what is:
- considered "start"
- what happens when the user misses a week

I would suggest completing just the first part of this with setting the date in the internal handlers and then the rest of this is blocked until we have a technical spec from @timoguin on how the state logic is supposed to work (unless we already have that somewhere and im just not aware)

File: src/service/tiki-service/internal-handlers.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export class InternalHandlers {
     if (!receipt.blinkReceiptId) {
       console.warn(`Receipt missing ID. Skipping: ${JSON.stringify(receipt)}`);
     } else {
-      this.store.receipt.add(receipt.blinkReceiptId).catch((error): void => {
+      this.store.receipt.add(receipt.blinkReceiptId, new Date(receipt.receiptDate?.value!)).catch((error): void => {
```
Comment (mike-audi) [inline src/service/tiki-service/internal-handlers.ts:57]: I think we want to use `receiptDateTime: number;` to avoid timezone issues.

**Review by mike-audi — CHANGES_REQUESTED** [review]

File: src/service/tiki-service/internal-handlers.ts (hunk 1 above)
Comment (MiroBenicio) [reply src/service/tiki-service/internal-handlers.ts]: updated!

**Comment by mike-audi** [comment]
State logic changes can be found here: https://github.com/tiki/apps-receipt-capacitor/issues/326

**Pull request merged**